enhancements/machine-api/converting-machine-api-to-cluster-api.md

```diff
@@ -0,0 +1,754 @@
+  - Trigger a MachineHealthCheck reconciliation using a Machine API authoritative MachineHealthCheck
+  - Ensure that Machines are deleted correctly
+  - Switch authority of MachineHealthCheck to Cluster API
+  - Trigger a second MachineHealthCheck reconciliation using the now Cluster API authoritative MachineHealthCheck
+  - Ensure that Machines are deleted correctly
+
+[summary-of-rules]: #summary-of-the-rules-outlined-above
+
+### Graduation Criteria
+
+The project will initially be introduced under a feature gate `MachineAPIMigration`.
+The new synchronisation operator will be deployed on all `TechPreviewNoUpgrade`/`CustomNoUpgrade` clusters,
+but will check for the presence of the above feature gate before operating.
+
+#### Dev Preview -> Tech Preview
+
+Initially the operator will be introduced behind a feature gate and only accessible with a `CustomNoUpgrade` feature set.
+Before promotion to the `TechPreviewNoUpgrade` feature set, we will ensure:
+- The operator has reached a minimal level of functionality to be end to end testable
+- Manual testing allows at least one platform with basic configuration to be converted between Cluster API and Machine API
+- Tests have been run to ensure the operator does not break the existing payload jobs
+- The operator can report its status via conditions on the Machine API cluster operator
+
+#### Tech Preview -> GA
+
+Once the operator has been released under the `TechPreviewNoUpgrade` feature set, we will continue to enhance the operator
+before marking it as GA:
+- Full E2E testing of the various behaviours described regarding Machine synchronisation
+- E2E testing of MachineSet and ControlPlaneMachineSet conversion
+- E2E testing of MachineHealthCheck workflows in mixed instance management clusters
+- At least one platform has sufficient conversion logic to be useful and allow users on the platform to start migration
+
+Note, we will not require 100% feature parity for promotion to GA.
+Some features supported by Machine API may be supported in conversion only in subsequent releases.
+
+#### Removing a deprecated feature
+
+No features will be deprecated as a part of this enhancement.
+
+### Upgrade / Downgrade Strategy
+
+#### On Upgrade
+
+When upgrading into a release with the new synchronisation operator,
+new Cluster API resources will be created and existing Machine API resources will be annotated.
+
+There should be no effect on the cluster as each of these operations should be a no-op.
+
+#### On Downgrade
+
+On downgrade, the synchronisation of resources will stop.
+To prevent both Machine API and Cluster API resources being reconciled,
+we will make sure that Cluster API controllers in the 4.N-1 release do not reconcile when a mirror resource exists with the same name.
+Independent of the authoritative annotation.
+
+This will mean that on downgrade all Machine management reverts to Machine API controllers.
```

Comment:
Can we prevent downgrade when CAPI->MAPI is no longer possible for some objects?

Reply:
Good question, a Downgrade is actually an Upgrade according to CVO, so, if we can detect that there's going go to be an issue with the direction of the upgrade, we could set the upgradeable false condition

Reply:
Coming back to this, no, we couldn't do this. We would block all upgrades, or none, which would just be problematic